File: Doxan/src/main/java/com/doxan/doxan/domain/model/Post.java
package com.doxan.doxan.domain.model;

import com.doxan.doxan.domain.model.enums.ReactionType;
import com.doxan.doxan.domain.model.enums.Visibility;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Map;

@Setter
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Post {
    private String id;
    private User user;
    private String targetId;
    private String content;
    private Visibility visibility;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Map<ReactionType, Long> reactionCounts;
    private boolean edited;
    private int shareCount;
}

File: frontend/types/models/post.ts
import { Media } from "@/types/models/media"
import type { User } from "@/types/models/user"
import { ReactionCounts } from "./reaction-counts"

export type Visibility = "PUBLIC" | "FRIENDS" | "PRIVATE"

export interface Post {
  id: string
  content: string
  user: User
  images?: Media[]
  videos?: Media[]
  docs?: Media[]
  visibility: Visibility
  reactionCounts: ReactionCounts
  isEdited: boolean
  shareCount: number
  createdAt: Date
  updatedAt?: Date
}
